tests/DummyServerLib/DummyServer.cpp

```diff
@@ -70,16 +73,20 @@ void DummyServer::activate() {
       }
     }
   }
-  if(_debug) std::cout << "echoing port " << _backportNode << std::endl; // DEBUG
+  if(_debug) {
+    std::cout << "echoing port " << _backportNode << std::endl;
+  }
 
-  // finally start the main loop, which accesses the serial port
+  // Finally start the main loop, which accesses the serial port.
   _stopMainLoop = false;
   _mainLoopThread = boost::thread([&]() { mainLoop(); });
 }
 
 void DummyServer::deactivate() {
-  // First stop the main thread which is accessing the SerialPort object
-  std::cout << "DEBUG!!! << DummyServer::deactivate() joining main thread." << std::endl;
+  // First stop the main threadi, which is accessing the SerialPort object.
+  if(_debug) {
+    std::cout << "DEBUG!!! << DummyServer::deactivate() joining main thread." << std::endl;
+  }
```

Comment:
This one is part of the hack of understanding the race conditions in the unified test. It has to stay active even if _debug is false. We will remove this before the release.